wayk_proto/src/message/connection_sequence/capabilities.rs

```diff
@@ -310,19 +310,73 @@ __flags_struct! {
     }
 }
 
-#[derive(Encode, Decode, Debug, Clone)]
+#[derive(Debug, Clone)]
 pub struct SystemCapset {
     pub flags: SystemCapsetFlags,
-    pub os_info: NowSystemOsInfo,
+    pub os_info: Option<NowSystemOsInfo>,
 }
 
 impl SystemCapset {
-    const NAME: &'static str = "NowSystem";
+    pub const NAME: &'static str = "NowSystem";
 
     pub fn new_os_info(os_info: NowSystemOsInfo) -> Self {
         Self {
             flags: SystemCapsetFlags::new_empty().set_os_info(),
-            os_info,
+            os_info: Some(os_info),
+        }
+    }
+}
+
+impl Encode for SystemCapset {
+    fn encoded_len(&self) -> usize {
+        let mut os_info_len = 0;
+        if let Some(os_info) = &self.os_info {
+            os_info_len = os_info.encoded_len();
+        }
```

Comment:
Prefer expressions over mutability
```suggestion
        let os_info_len = if let Some(os_info) = &self.os_info {
            os_info.encoded_len()
        } else {
            0
        }
```